### PR TITLE
Switch deprecation method

### PR DESCRIPTION
**Pull request by levitte**
Done for most headers.  I've avoided those that I know are in the works in other PRs (the DH, DSA and EC deprecation PRs)

### CODE REVIEW

File: include/openssl/aes.h [hunk 1]
```diff
@@ -45,68 +45,67 @@ struct aes_key_st {
...
+                        size_t length, const AES_KEY *key,
+                        unsigned char *ivec, int *num);
+# endif
 
 /* NB: the IV is _two_ blocks long */
-DEPRECATEDIN_3_0(void
-                 AES_ige_encrypt(const unsigned char *in, unsigned char *out,
-                                 size_t length, const AES_KEY *key,
-                                 unsigned char *ivec, const int enc))
+# ifndef OPENSSL_NO_DEPRECATED_3_0
```
Comment (h-vetinari) [inline include/openssl/aes.h:88]: Closing and reopening the `# ifndef` should not be necessary for just the comment?

File: include/openssl/aes.h [hunk 2]
```diff
@@ -45,68 +45,67 @@ struct aes_key_st {
...
+OSSL_DEPRECATEDIN_3_0
+void AES_ige_encrypt(const unsigned char *in, unsigned char *out,
+                     size_t length, const AES_KEY *key,
+                     unsigned char *ivec, const int enc);
+# endif
 /* NB: the IV is _four_ blocks long */
```
Comment (h-vetinari) [inline include/openssl/aes.h:94]: Same here

File: include/openssl/aes.h (hunk 1 above)
Comment (levitte) [reply include/openssl/aes.h:88]: Thanks.  I didn't notice that one.  `util/fix-deprecation` is only so smart

**Comment by levitte** [comment]
Rebased for GH action fixups

**Review by t8m — APPROVED** [review]
Great work!

**Comment by levitte** [comment]
Merged

0016a034a7da28df038c9b4a5fd0417b0d9a52af Switch deprecation method for AES
d1b99dd905b2405c77243fcf17f3390512d7731a Switch deprecation method for ASN.1
894fe6e1d68216b8c656c4f5e6cde157747b7ae0 Switch deprecation method for BIO
6638749f0f168d5219ae311bcf14048ff4dd4429 Switch deprecation method for Blowfish
4b66e5256f515c53e95c55a1598862ec529cb5f8 Switch deprecation method for BIGNUM
0294097253fcdafb97085de70a757b908e346bf2 Switch deprecation method for Camellia
ebafdfac737e1f49ac1ed5307edff99c061c82a1 Switch deprecation method for CAST
db617a4a6bdd35baadc62af84eca6bb698b55677 Switch deprecation method for CMAC
7f7c856c66709a419136b6bf4cde7ef0bf7b7ede Switch deprecation method for CONF
05cb22388ffe71b59c048c94c90affe4413d3f2b Switch deprecation method for CRYPTO
27147678ca7779d20fef5bb93a4d1cd0abccc573 Switch deprecation method for DES
d43788bd207dd7e9ff0feb9dc981496b217809bb Switch deprecation method for ENGINE
f6be9ae210669b505e6ab8300d17f819658551e9 Switch deprecation method for ERR
306265819b106c27251d66f41ecc8f572ad66ced Switch deprecation method for EVP
ba21405888340f1e2038939468669b9c1bc0313b Switch deprecation method for HMAC
4b361f87a98a3ad2f78eee9308dfc56c428d302b Switch deprecation method for IDEA
0c5f728456e5bb726d14f6423d120e4d01da661d Switch deprecation method for MD2
4ea7e4e013e8159816cfb9928460a8f3f6044a1d Switch deprecation method for MD4
5fb0f3178fc23739781c5ba219180e64a82a2314 Switch deprecation method for MD5
be5b2b64a559628dff876cfdf028308cbe6c7e2d Switch deprecation method for MDC2
be8e9b39a32d6eae59f8b7eab48e9535bd4e9bb2 Switch deprecation method for PKCS#12
00639486b0ab66d9cd89ab47fc56a3ac6cb40084 Switch deprecation method for RAND
1b6d1f51b40d95bc6bd6fae67f906ca856b1aa7a Switch deprecation method for RC2
53a14afee9a2a595fd567f9fefdbeae43fe02c75 Switch deprecation method for RC4
17fbbe7727f62d288c5ade9576228fce5390714a Switch deprecation method for RC5
632fb5259f066b1b3a5971905da350a1743cde84 Switch deprecation method for RIPEMD
0038fff07d39515acfbd123b85c45c877350db0a Switch deprecation method for SEED
033b8d5ee0863228718aa17c241f2adea30092d5 Switch deprecation method for SHA
2b3f59fd494145fffa41943b003025887e8aec7d Switch deprecation method for SRP
3be1dc50aa9d5394fd545adf792c17ccc3dbaec5 Switch deprecation method for SSL
f2e94543641f45ace6e960cd7d8592bf537fb55d Switch deprecation method for OSSL_STORE
3c667d9888bc719169e74cb797a0783da3893122 Switch deprecation method for Whirlpool
76191c7999e0d1f709ea468950457f71cea378c4 Switch deprecation method for X.509

**Comment by richsalz** [comment]
Thanks for doing the work to convert everything to be consistent!

**Comment by levitte** [comment]
Not entirely everything. I didn't touch DSA and EC yet, since I know there are two other PRs in the works that already deal with deprecations for those types.

**Comment by alorbach** [comment]
May I ask why deprecation was switch for all those API's?
I see a lot of depreciated warnings when using ENGINE_* API functions but even the documentation samples are using them, so what would be the alternatives?

**Comment by t8m** [comment]
Please read https://www.openssl.org/docs/man3.0/man7/migration_guide.html

**Pull request closed without merging**